File: solutions/ruby/2022/01/test/solutions_test.rb
# frozen_string_literal: true

require 'bundler/setup'
require 'minitest/autorun'
require 'minitest/pride'
require_relative '../lib/solutions'

class TestSolutions202201 < Minitest::Test
  INPUT = File.read File.join(__dir__, '../../../../../inputs/2022/01/test_input')

  def test_part_1
    assert_equal 24000, Solutions202201.part_1(INPUT)
  end

  def test_part_2
    assert_equal 45000, Solutions202201.part_2(INPUT)
  end
end

File: solutions/elixir/2022/01/test/solutions_test.exs
Code.require_file("../lib/solutions.exs", __DIR__)

defmodule Solutions202201Test do
  use ExUnit.Case

  @input ~S"""
  1000
  2000
  3000

  4000

  5000
  6000

  7000
  8000
  9000

  10000
  """

  test ".part_1" do
    assert Solutions202201.part_1(@input) == 24000
  end

  test ".part_2" do
    assert Solutions202201.part_2(@input) == 45000
  end
end
